department-of-veterans-affairs/caseflow
# [Part 1] Add Special Issues to Legacy Appeals

Connected to #14269 and #13396
Resolves #14269 

PART 1 of the stack
PART 2: #14425 

### Description
Finalize Special Issue implementation in Caseflow Dispatch

 - Adds new specialIssues to LegacyAppeal model, allowing their addition
    at dispatch (sets allowed params)
      - added note about this to SpecialIssues.jsx
  - bugfix: corrects FeatureToggle name from snake case to camel case, setting
     the initial state in the component

### Acceptance Criteria

### Testing Plan

Migrate your database!
`make update`

State Bug:
1. On master, log in as a dispatch user
1. Go to `dispatch/establish-claim` and click `Establish Next Claim`
1. Open your React inspector and confirm the state is **missing** the new special issues (blue_water, burn_pit, cavc, MST, and no_special_issues)

Now checkout this branch, perform the same steps, but see that the initial state has both the original and the new special issues.

Dispatch:
1. On master, log in as a dispatch user
1. Go to `dispatch/establish-claim` and click `Establish Next Claim` - note the id in the URL
1. Choose all of the new special issues and continue through dispatch. 
1. Confirm dispatch succeeds
1. Confirm the LegacyAppeal has all the new specials issue saved.
```ruby
dispatchtask_id = 
appeal_id = Dispatch::Task.find(dispatchtask_id).appeal_id
LegacyAppeal.find(appeal_id).special_issues
```

### Database Changes
*Only for Schema Changes*

* ~Timestamps (created_at, updated_at) for new tables~
* [x] DB schema docs updated with `make docs`
